queue.go

```diff
@@ -260,6 +281,35 @@ func (q *DQue) Peek() (interface{}, error) {
 	return obj, nil
 }
 
+// BatchPeek returns a slice of up to 1-n objects without dequeueing them.
+// Fewer than n items may be returned, depending on the remaining objects in the first segment.
+// ErrEmpty is returned if the queue is empty.
+func (q *DQue) BatchPeek(n int) ([]interface{}, error) {
+	var objects []interface{}
+
+	// Lock the entire queue to get a consistent slice of objects.
+	// note: this could potentially be optimized in the future to only lock the first segment.
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+
+	for i := 0; i < n; i++ {
+		obj, err := q.firstSegment.peek(i)
+		if err != nil {
+			if err == errIndexOutOfRange {
+				if len(objects) > 0 {
+					return objects, nil
+				}
+				return nil, ErrEmpty
+			}
+			return nil, err
+		}
+
+		objects = append(objects, obj)
+	}
+
+	return objects, nil
+}
+
```

Comment:
Thank you for tackling this.  Consider the situation where the first segment only has one item left in it while the second segment is full.  If someone calls BatchPeek(5), this will return only 1 item because the remaining 4 are in the second segment.  Returning all 5 could get messy and use more memory because you'd need to have more than just the first and last segments open... and potentially more than one of the middle segments open if they are peeking for a lot of items.